src/main/java/org/alfresco/heartbeat/HeartBeatJob.java

```diff
@@ -0,0 +1,198 @@
+import org.quartz.JobDataMap;
+import org.quartz.JobExecutionContext;
+import org.quartz.JobExecutionException;
+
+import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * The scheduler job responsible for triggering a heartbeat on a regular basis.
+ */
+
+public class HeartBeatJob implements Job
+{
+    /** The logger. */
+    private static final Log logger = LogFactory.getLog(HeartBeatJob.class);
+
+    /** The heart beat job will be locked for max 30 seconds */
+    private static final long LOCK_TTL = 30000L;
+
+    public void execute(final JobExecutionContext jobexecutioncontext) throws JobExecutionException
+    {
+        final JobDataMap dataMap = jobexecutioncontext.getJobDetail().getJobDataMap();
+        final HBBaseDataCollector collector = (HBBaseDataCollector) dataMap.get("collector");
+        final HBDataSenderService hbDataSenderService = (HBDataSenderService) dataMap.get("hbDataSenderService");
+        final JobLockService jobLockService = (JobLockService) dataMap.get("jobLockService");
+
+        if(collector == null)
+        {
+            if (logger.isDebugEnabled())
+            {
+                logger.debug("Exit HeartBeatJob because there is no assigned HB collector");
+            }
+            return;
+        }
+        if(hbDataSenderService == null)
+        {
+            if (logger.isDebugEnabled())
+            {
+                logger.debug("Exit HeartBeatJob because there is no HBDataSenderService");
+            }
+            return;
+        }
+        if(jobLockService == null)
+        {
+            if (logger.isDebugEnabled())
+            {
+                logger.debug("Exit HeartBeatJob because there is no JobLockService");
+            }
+            return;
+        }
+        QName qName = QName.createQName(NamespaceService.SYSTEM_MODEL_1_0_URI, collector.getCollectorId());
+        String lockToken = null;
+        LockCallback lockCallback = new LockCallback(qName);
+        try
+        {
+            // Get a dynamic lock
+            lockToken = acquireLock(lockCallback, qName, jobLockService);
+            collectAndSendDataLocked(collector, hbDataSenderService);
+        }
+        catch (LockAcquisitionException e)
+        {
+            if (logger.isDebugEnabled())
+            {
+                logger.debug("Skipping collect and send data (could not get lock): " + e.getMessage());
+            }
+        }
+        finally
+        {
+            releaseLock(lockCallback, lockToken, qName, jobLockService);
+        }
+    }
+
+    private void collectAndSendDataLocked(final HBBaseDataCollector collector, final HBDataSenderService hbDataSenderService) throws JobExecutionException
+    {
+        try
+        {
+            List<HBData> data = collector.collectData();
+            try
+            {
+                hbDataSenderService.sendData(data);
+            }
+            catch (Exception e)
+            {
+                logger.error(e);
```

Comment:
Is there a reason why we are logging the exception from the sender as error but as debug/warn when it is from the collector?

Reply:
Just checked and saw that that is how it was done before..